src/app/shared/modal/modal-login/modal-login.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { AuthenticationService } from '../../services/authentication.service';
+import { User } from '../../interfaces/user.interface.';
+
+@Component({
+  selector: 'app-modal-login',
+  templateUrl: './modal-login.component.html',
+  styleUrls: ['./modal-login.component.css']
+})
+export class ModalLoginComponent implements OnInit {
+  @Input() appHideIfAuthenticated!: boolean;
+  @Output() userLogged: EventEmitter<void> = new EventEmitter<void>();
+
+  email!: string;
+  password!: string;
+  isButtonEnabled: boolean = false;
+  id: number = 1;
+
+  constructor(private authService: AuthenticationService) { }
+
+  ngOnInit() {
+    if (this.email && this.password) {
+      this.isButtonEnabled = true;
+    }
+    else {
+      this.isButtonEnabled = false;
+    }
+  }
+
+  login(): void {
+    const newUser: User = {
+      id: this.id,
```

Comment:
Id is usually not passed from the frontend when login
It can be return from backend however :)

Reply:
Ok, understood, I think delete it in Http part of the course)